src/app/core/form-dialog/dialog-buttons/dialog-buttons.component.ts

```diff
@@ -45,8 +48,20 @@ export class DialogButtonsComponent implements OnInit {
     private alertService: AlertService,
     private entityRemoveService: EntityRemoveService,
     private router: Router,
-    private ability: EntityAbility
-  ) {}
+    private ability: EntityAbility,
+    private confirmation: ConfirmationDialogService
+  ) {
+    this.dialog.backdropClick().subscribe(() => {
+      if (this.form.dirty) {
+        this.confirmation
+          .getConfirmation(
+            $localize`:Save changes header:Save Changes?`,
+            $localize`:Save changes message:Do you want to save the changes you made?`
```

Comment:
This could also follow the same style as the popup for navigation where its about discarding instead of saving.

Reply:
good point, would be great to have the same pattern everywhere! (Maybe even define a default text in a method in confirmation service already to have it 100% consistent?)